Terminal.Gui/Input/Keyboard.cs

```diff
@@ -0,0 +1,661 @@
+using System;
+using System.Text;
+
+namespace Terminal.Gui;
+
+/// <summary>
+/// Defines the event arguments for keyboard events.
+/// </summary>
+public class KeyEventArgs : EventArgs {
+
+	/// <summary>
+	/// Constructs a new <see cref="KeyEventArgs"/>
+	/// </summary>
+	public KeyEventArgs () : this (Key.Unknown) { }
+
+	/// <summary>
+	///   Constructs a new <see cref="KeyEventArgs"/> from the provided Key value - can be a rune cast into a Key value
+	/// </summary>
+	/// <param name="k">The key</param>
+	public KeyEventArgs (Key k)
+	{
+		Key = k;
+	}
+
+	/// <summary>
+	/// Indicates if the current Key event has already been processed and the driver should stop notifying any other event subscriber.
+	/// Its important to set this value to true specially when updating any View's layout from inside the subscriber method.
+	/// </summary>
+	public bool Handled { get; set; } = false;
+
+	/// <summary>
+	/// Symbolic definition for the key.
+	/// </summary>
+	public Key Key { get; set; }
+
+	/// <summary>
+	///   The key value cast to an integer, you will typical use this for
+	///   extracting the Unicode rune value out of a key, when none of the
+	///   symbolic options are in use.
+	/// </summary>
+	public int KeyValue => (int)Key;
+
+	/// <summary>
+	/// Gets a value indicating whether the Shift key was pressed.
+	/// </summary>
+	/// <value><see langword="true"/> if is shift; otherwise, <see langword="false"/>.</value>
+	public bool IsShift => (Key & Key.ShiftMask) != 0;
+
+	/// <summary>
+	/// Gets a value indicating whether the Alt key was pressed (real or synthesized)
+	/// </summary>
+	/// <value><see langword="true"/> if is alternate; otherwise, <see langword="false"/>.</value>
+	public bool IsAlt => (Key & Key.AltMask) != 0;
+
+	/// <summary>
+	/// Gets a value indicating whether the Ctrl key was pressed.
+	/// </summary>
+	/// <value><see langword="true"/> if is ctrl; otherwise, <see langword="false"/>.</value>
+	public bool IsCtrl => (Key & Key.CtrlMask) != 0;
+
+	/// <summary>
+	/// Pretty prints the KeyEvent
+	/// </summary>
+	/// <returns></returns>
+	public override string ToString ()
+	{
+		return ToString (Key, (Rune)'+');
+	}
+
+	private static string GetKeyString (Key key)
+	{
+		if (key is Key.Null or Key.SpecialMask) {
+			return string.Empty;
+		}
+
+		bool lowerCase = false;
+		if (key > Key.Space && key < Key.D1) {
+			key = (Key)(key - Key.Space);
+			lowerCase = true;
+		}
```

Comment:
This should be removed because `Key > Key.Space` are valid keys. Is the case of the `[InlineData((Key)'\'', "\'")]`